internal/cmd/skupper/debug/nonkube/debug.go

```diff
@@ -21,13 +45,355 @@ func NewCmdDebug() *CmdDebug {
 }
 
 func (cmd *CmdDebug) NewClient(cobraCommand *cobra.Command, args []string) {
+	if cmd.CobraCmd != nil && cmd.CobraCmd.Flag(common.FlagNameNamespace) != nil && cmd.CobraCmd.Flag(common.FlagNameNamespace).Value.String() != "" {
+		cmd.namespace = cmd.CobraCmd.Flag(common.FlagNameNamespace).Value.String()
+	}
+
+	cmd.connectorHandler = fs.NewConnectorHandler(cmd.namespace)
+	cmd.listenerHandler = fs.NewListenerHandler(cmd.namespace)
+	cmd.siteHandler = fs.NewSiteHandler(cmd.namespace)
+	cmd.linkHandler = fs.NewLinkHandler(cmd.namespace)
+	cmd.routerAccessHandler = fs.NewRouterAccessHandler(cmd.namespace)
+	cmd.certificateHandler = fs.NewCertificateHandler(cmd.namespace)
+	cmd.secretHandler = fs.NewSecretHandler(cmd.namespace)
+	cmd.configMapHandler = fs.NewConfigMapHandler(cmd.namespace)
+}
+
+func (cmd *CmdDebug) ValidateInput(args []string) error {
+	var validationErrors []error
+	fileStringValidator := validator.NewFilePathStringValidator()
+
+	// Validate dump file name
+	if len(args) < 1 {
+		cmd.fileName = "skupper-dump"
+	} else if len(args) > 1 {
+		validationErrors = append(validationErrors, fmt.Errorf("only one argument is allowed for this command"))
+	} else if args[0] == "" {
+		validationErrors = append(validationErrors, fmt.Errorf("filename must not be empty"))
+	} else {
+		ok, err := fileStringValidator.Evaluate(args[0])
+		if !ok {
+			validationErrors = append(validationErrors, fmt.Errorf("filename is not valid: %s", err))
+		} else {
+			cmd.fileName = args[0]
+		}
+	}
 
+	return errors.Join(validationErrors...)
 }
 
-func (cmd *CmdDebug) ValidateInput(args []string) error { return nil }
+func (cmd *CmdDebug) InputToOptions() {
+	datetime := time.Now().Format("20060102150405")
+	cmd.fileName = fmt.Sprintf("%s-%s-%s", cmd.fileName, cmd.namespace, datetime)
+}
+
+func (cmd *CmdDebug) Run() error {
+	dumpFile := cmd.fileName
+
+	// Add extension if not present
+	if filepath.Ext(dumpFile) == "" {
+		dumpFile = dumpFile + ".tar.gz"
+	}
+
+	tarFile, err := os.Create(dumpFile)
+	if err != nil {
+		return fmt.Errorf("Unable to save skupper dump details: %w", err)
+	}
+
+	// compress tar
+	gz := gzip.NewWriter(tarFile)
+	defer gz.Close()
+	tw := tar.NewWriter(gz)
+	defer tw.Close()
+
+	platform := config.GetPlatform()
```

Comment:
The platform, for an active namespace, should be loaded using:

```golang
	nsl := &NamespacePlatformLoader{}
	platform, err := nsl.Load("your-namespace")
```

Reply:
done